src/core/mac/mac_frame.hpp

```diff
@@ -256,57 +256,43 @@ class Address
      */
     typedef ExtAddress::InfoString InfoString;
 
-    /**
-     * This enumeration specifies the IEEE 802.15.4 Address type.
-     *
-     */
-    enum Type
```

Comment:
Suggeston: Keep this definition as is. 
Then convert from this type to `otMacAddress` (which I suggested to move out `radio.h` and into `util/mac_frame.h`) in util/mac-frame code.

Reply:
no problem, thanks for the suggestions

Reply:
@abtink if we don't let Mac::Address inherit `otMacAddress`, we have to use `reinterpret_cast`, are you OK with it?

Reply:
If we can ensure it works correctly. Alternative can be to convert and re-pooluate the values/feidls accordingly.